src/shared/sol-util.c

```diff
@@ -312,14 +315,37 @@ sol_util_str_split(const struct sol_str_slice slice, const char *delim, size_t m
 static int
 get_progname(char *out, size_t size)
 {
-    char cwd[PATH_MAX] = { NULL }, *execfn;
+    char cwd[PATH_MAX] = { NULL }, *execfn = NULL;
 
     if (getcwd(cwd, sizeof(cwd)) == NULL)
-        return -1;
+        return -errno;
 
+#ifdef HAVE_SYS_AUXV_H
     execfn = (char *)getauxval(AT_EXECFN);
```

Comment:
Why won't you check for the symbol instead only check for the header (same thing is done for accept4), and if it's not defined create it on sol-missing ? 

I don't see where it is different from the strndupa and accept4 cases ?